tests/acceptance/EditSignUpConfirmationEmailCest.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace MailPoet\Test\Acceptance;
+
+use Codeception\Util\Locator;
+
+class EditSignUpConfirmationEmailCest {
+
+  function edit(\AcceptanceTester $I) {
+    $I->wantTo('Edit sign up confirmation email');
+
+    $I->login();
+    $I->amOnMailPoetPage('Settings');
+    $I->click('[data-automation-id="signup_settings_tab"]');
+    $I->waitForText('Enable sign-up confirmation');
+
+    // make sure sign up confirmation is enabled
+    $I->click('[data-automation-id="enable_signup_confirmation"]');
+    $I->acceptPopup();
+
+    // edit confirmation email
+    $I->fillField('[data-automation-id="signup_confirmation_email_from_name"]', 'Confirmation Test From');
+    $I->fillField('[data-automation-id="signup_confirmation_email_from_email"]', 'from-confirmation-test@example.com');
+    $I->fillField('[data-automation-id="signup_confirmation_email_subject"]', 'Confirmation email subject');
+    $I->fillField('[data-automation-id="signup_confirmation_email_body"]', 'Confirmation email body [activation_link]link[/activation_link]');
+
+    $I->click('[data-automation-id="settings-submit-button"]');
+
+    $I->createFormAndSubscribe();
+
+    // check the received email
+    $I->amOnUrl(\AcceptanceTester::MAIL_URL);
+    $I->waitForText('Confirmation email subject');
+    $I->click(Locator::contains('span.subject', 'Confirmation email subject'));
+
+    $I->waitForText('Confirmation email subject');
+    $I->waitForText('Confirmation Test From <from-confirmation-test@example.com>');
+    $I->switchToIframe('preview-html');
+    $I->waitForText('Confirmation email body link');
+  }
+
+  function _after(\AcceptanceTester $I) {
+    $I->cli('widget reset sidebar-1 --allow-root');
+    $I->amOnUrl(\AcceptanceTester::WP_URL);
```

Comment:
The cleanup here looks like quite a complicated process. What about adding some SettingsFactory or some similar class (to wrap a model) which will be able to get or set some settings directly to database. I think this could be useful if we will need to prepare some specific settings within other tests. What do you think?

Reply:
Sounds like a good idea. Do you think that should be part of this story, or should that be a new story that would clean this code as well?

Reply:
I think it can be an extra issue